File: AngularFrontEnd/src/app/pojos/AuthResponse.ts
import { UserBase } from './UserBase';
import { User } from './User';
import { Driver } from './Driver';
import { Admin } from './Admin';
import { SuperAdmin } from './SuperAdmin';

export class AuthResponse {
  authenticated?: boolean;
  userDetails?: UserBase | User | Driver | Admin | SuperAdmin;

  constructor(authenticated: boolean, userBase: UserBase);
  constructor(authenticated: boolean, user: User);
  constructor(authenticated: boolean, driver: Driver);
  constructor(authenticated: boolean, admin: Admin);
  constructor(authenticated: boolean, superAdmin: SuperAdmin);

  constructor(authenticated: boolean, userDetails: UserBase | User | Driver | Admin | SuperAdmin) {
    this.authenticated = authenticated;
    this.userDetails = userDetails;
  }
}

File: SpringBootProject/src/main/java/com/caltech/pojo/AuthResponse.java
package com.caltech.pojo;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class AuthResponse {
    private boolean authenticated;
    private Object userDetails;
    
    public AuthResponse(boolean authenticated, UserBase userBase) {
        this.authenticated = authenticated;
        this.userDetails = userBase;
    }

    public AuthResponse(boolean authenticated, Driver driver) {
        this.authenticated = authenticated;
        this.userDetails = driver;
    }

    public AuthResponse(boolean authenticated, Admin admin) {
        this.authenticated = authenticated;
        this.userDetails = admin;
    }
}

